source/storage.rst

```diff
@@ -446,6 +446,10 @@ the VM is running.
    running when the live migration operation is requested.
 
 
+For KVM, live storage migration is available from the 4.11 release
+and currently only supports migration from NFS/CEPH to SolidFire Managed Storage.
```

Comment:
live migration of volumes between NFS-NFS or CEPH-CEPTH is not working?

Reply:
Correct there is actually no thing as live storage migration on KVM, until 4.11 - we were working with SolidFire and Mike Tutkowski, and requested this feature, which is now made available in community. Only OFFLINE storage migration (nfs/ceph to nfs/ceph and now (4.11) also to SF works)

Reply:
But Mike's code can and should be perhaps generalizes to support NFS<-->NFS, NFS<-->CEPH, and CEPH <--> CEPH. Because at the end it boils down to:
- precreating empty volumes on new storage (destination)
-  dumping and editing XML file to point to new volumes on new storage (rest of XML is identical, except also VNC IP listen address)
- then it's just a libvirt call following "virsh migrate --live --persistent --copy-storage-all --verbose --xml i-2-18-VM.xml i-2-18-VM qemu+ssh://10.117.40.112/system"

I have no development skills, and small inhouse team, but would otherwise be very happy to see such feature (not that we actually need it any more...)

Reply:
Thanks for the explanation. You are already helping documenting it, and that is already a great job! Do not worry about features that do not work yet. I only asked because I thought that the live migration of volumes between primary storage of the same type was already supported in KVM.